src/CacheTrait.php

```diff
@@ -62,7 +62,17 @@ private function setCachedValue($v)
         if (is_null($fetcherKey)) {
             return;
         }
-        $key = $this->cacheConfig['prefix'].$fetcherKey;
-        $this->cache->set($key, $v);
+
+        $key = self::getValidKeyName($this->cacheConfig['prefix'].$fetcherKey);
+        $cacheItem = $this->cache->getItem($key);
+        $cacheItem->set($v);
+        $cacheItem->expiresAfter($this->cacheConfig['lifetime']);
+        return $this->cache->save($cacheItem);
+    }
+
+    public static function getValidKeyName($key)
```

Comment:
@kesar you're right, this function's visibility should match that of `getCachedValue` and `setCachedValue`